pkg/spinlock/wait.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package spinlock
+
+import (
+	"errors"
+	"testing"
+	"time"
+)
+
+var ErrTimedOut = errors.New("timed out waiting for condition")
+
+// Wait blocks execution until condition is satisfied or until it times out.
+func Wait(t *testing.T, timeoutDur time.Duration, cond func() bool) error {
```

Comment:
since we use this only for testing, it's not ideal to have this under the root pkg folder.

Reply:
How to deal with this in that case?